### PR TITLE
Add tcmalloc for amd64 platform

### PR DESCRIPTION
**Pull request by derekbit**
Depends on https://github.com/rancher/tgt/pull/8 and https://github.com/rancher/liblonghorn/pull/16

[Longhorn 4136](https://github.com/longhorn/longhorn/issues/4136)

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

**Review by innobead — DISMISSED** [review]
LGTM

After discussing, we have two things here.
- Supporting arm64 is probably doable, but right now we only focus on amd64 first (need some testing for arm64
- Just use tcmalloc lib from gperftools installation.

**Comment by innobead** [comment]
BTW, is this verifiable? Could we understand any obvious performance difference after this introduction? If yes, that will be good.

**Comment by innobead** [comment]
@derekbit https://github.com/rancher/tgt/pull/8 and https://github.com/rancher/liblonghorn/pull/16 all merged. Do you think this's ready for review?

File: package/Dockerfile [hunk 1]
```diff
@@ -7,6 +7,10 @@ RUN zypper -n addrepo --refresh https://download.opensuse.org/repositories/syste
     zypper -n addrepo --refresh https://download.opensuse.org/repositories/network:utilities/SLE_15_SP3/network:utilities.repo && \
     zypper --gpg-auto-import-keys ref
 
+RUN if [ ${ARCH} == "amd64" ]; then \
+        zypper -n install autoconf libtool libunwind-devel gcc-c++; \
```
[the remark this answers is not in the document]
Comment (derekbit) [reply package/Dockerfile:11]: Addressed. Thank you.

File: package/Dockerfile [hunk 2]
```diff
@@ -16,23 +20,48 @@ RUN zypper -n install cmake kmod curl nfs-client nfs4-acl-tools fuse git gcc \
...
+        git checkout db6d94d9893fdd8c74028b5113cd9185be1e3111 && \
+        ./autogen.sh && \
+        ./configure && \
+        make && \
+        make install; \
+        zypper -n remove autoconf libtool libunwind-devel gcc-c++; \
+        rm -rf /usr/src/gperftools; \
+    fi
+
+RUN ldconfig
```
Comment (innobead) [inline package/Dockerfile:40]: Suggest moving ldconfig to the end of the previous RUN to avoid an extra layer in the final image.

**Review by innobead — CHANGES_REQUESTED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@derekbit need to backport to 1.3.1. Thanks.

**Pull request merged**